packages/front-end/pages/product-analytics/dashboards/new.tsx

```diff
@@ -0,0 +1,137 @@
+import React, { useState, useCallback } from "react";
+import { useRouter } from "next/router";
+import { DashboardInterface } from "back-end/src/enterprise/validators/dashboard";
+import { useDefinitions } from "@/services/DefinitionsContext";
+import { useUser } from "@/services/UserContext";
+import { useAuth } from "@/services/auth";
+import DashboardWorkspace from "@/enterprise/components/Dashboards/DashboardWorkspace";
+import {
+  SubmitDashboard,
+  UpdateDashboardArgs,
+} from "@/enterprise/components/Dashboards/DashboardsTab";
+import PremiumCallout from "@/ui/PremiumCallout";
+import { useDashboards } from "@/hooks/useDashboards";
+
+export default function NewDashboardPage() {
+  const { project } = useDefinitions();
+  const { userId, hasCommercialFeature } = useUser();
+  const { apiCall } = useAuth();
+  const router = useRouter();
+  const { mutateDashboards } = useDashboards(false);
+
+  // Create a temporary dashboard object for the new dashboard
+  const createTemporaryDashboard = useCallback((): DashboardInterface => {
+    const now = new Date();
+    return {
+      id: "new",
+      uid: "new",
+      organization: "", // Will be set by backend
+      experimentId: "",
+      isDefault: false,
+      isDeleted: false,
+      userId: userId || "",
+      editLevel: "private",
+      shareLevel: "private",
+      enableAutoUpdates: false,
+      title: "Untitled Dashboard",
+      blocks: [],
+      projects: project ? [project] : [],
+      dateCreated: now,
+      dateUpdated: now,
+    };
+  }, [project, userId]);
+
+  const [dashboard, setDashboard] = useState<DashboardInterface>(
+    createTemporaryDashboard,
+  );
+
+  const handleSubmitDashboard: SubmitDashboard<UpdateDashboardArgs> =
+    useCallback(
+      async (args) => {
+        console.log("args", args);
+        // If dashboardId is "new", we need to create the dashboard (POST)
+        if (args.dashboardId === "new") {
+          const res = await apiCall<{
+            status: number;
+            dashboard: DashboardInterface;
+          }>("/dashboards", {
+            method: "POST",
+            body: JSON.stringify({
+              title: dashboard.title,
+              editLevel: dashboard.editLevel,
+              shareLevel: dashboard.shareLevel,
+              enableAutoUpdates: dashboard.enableAutoUpdates,
+              experimentId: "",
+              projects: dashboard.projects || [],
+              blocks: args.data.blocks || dashboard.blocks,
+            }),
+          });
+
+          if (res.status === 200) {
+            setDashboard(res.dashboard);
+          } else {
+            console.error(res);
+            throw new Error("Failed to create dashboard");
+          }
```

Comment:
Nit: This if/else is not needed.  `apiCall` throws if the response is non-200, so if it reaches here you can assume it succeeded.  Same with the one below.